packages/conform-dom/submission.ts

```diff
@@ -247,7 +247,25 @@ export function replySubmission<FormError>(
 		(context.intent?.type === 'reset' &&
 			(context.intent.payload.name ?? '') === '')
 	) {
-		return { initialValue: null };
+		const extraError =
+			'formErrors' in options || 'fieldErrors' in options
+				? normalize<Record<string, FormError | null>>({
+						'': options.formErrors ?? null,
+						...options.fieldErrors,
+					})
+				: null;
+		const error =
+			context.error || extraError
+				? {
+						...context.error,
+						...extraError,
+					}
+				: undefined;
```

Comment:
@edmundhung I'm not sure if it makes sense to show the error status, but I kept it in because I'm not sure what the expectations are